sql/ddl_api.sql

```diff
@@ -93,24 +95,44 @@ BEGIN
         FROM hypertable_ids INNER JOIN LATERAL _timescaledb_internal.dimension_get_time(hypertable_ids.id) time_dim ON (true);
     EXCEPTION
         WHEN NO_DATA_FOUND THEN
+        -- Q:: show chunks implementation now does this, shall I get rid of this transaction alltogether?
             RAISE EXCEPTION 'no hypertables found';
         WHEN TOO_MANY_ROWS THEN
             RAISE EXCEPTION 'cannot use drop_chunks on multiple tables with different time types';
     END;
 
 
     IF time_type = 'TIMESTAMP'::regtype THEN
-        PERFORM @extschema@.drop_chunks((now() - older_than)::timestamp, table_name, schema_name, cascade);
+        PERFORM @extschema@.drop_chunks((now() - older_than)::timestamp, table_name, schema_name, cascade,
+        (now() - newer_than)::timestamp);
     ELSIF time_type = 'DATE'::regtype THEN
-        PERFORM @extschema@.drop_chunks((now() - older_than)::date, table_name, schema_name, cascade);
+        PERFORM @extschema@.drop_chunks((now() - older_than)::date, table_name, schema_name, cascade,
+        (now() - newer_than)::date);
     ELSIF time_type = 'TIMESTAMPTZ'::regtype THEN
-        PERFORM @extschema@.drop_chunks(now() - older_than, table_name, schema_name, cascade);
+        PERFORM @extschema@.drop_chunks(now() - older_than, table_name, schema_name, cascade,
+        now() - newer_than);
     ELSE
         RAISE 'can only use drop_chunks with an INTERVAL for TIMESTAMP, TIMESTAMPTZ, and DATE types';
     END IF;
 END
 $BODY$;
 
+-- show chunks older than or newer than a specific time.
+-- hypertable_name can be a valid hypertable or NULL.
+-- In the latter case the function will try to list all
+-- the chunks from all of the hypertables in the database.
+-- older_than or newer_than or both can be NULL.
+-- if hypertable_name is null but a time constraint is specified
+-- through older_than or newer_than, the call will succeed
+-- if and only if all the hypertables in the database
```

Comment:
I'm not sure why this needs to be true. In `chunk_show` below, it looks like you have a foreach that loops through every ht and does this type-checking. If that's case, why can't you just ignore hypertables that don't have the correct type and show chunks for valid hypertables? In particular, instead of erroring out as soon as you hit a hypertable that doesn't typecheck, just have a `continue`?

It looks like the same is true for the SQL function `time_dim_type_check`, but I guess this makes more sense because you're in SQL..?

Reply:
Well, the reason I implemented it that way is because drop_chunks worked similarly before.
Maybe we could discuss this option too. It does provide a nice feature. For example, if the user has a set of tables with bigint time type and another set with timestamp, this allow to easily select one set or the other.

However, it can be a source of some caveats too. For example, apparently people often mix up TIMESTAMP and TIMESTAMPTZ specially when using now() or sth similar to create these (I have been there way too many times!!) So, I think throwing an error is safer.

Also, there is no *documented* way to tell which tables are the chunks returned by `show_chunks` from. So, I would rather not give users more opportunities to think they are dealing with one set of tables when they are dealing with another.